crates/bin/prove_block/src/main.rs

```diff
@@ -321,146 +387,113 @@ async fn main() -> Result<(), Box<dyn Error>> {
-                            sierra_program_debug_info: None,
-                            abi: None,
-                        };
-                        let casm_cc =
-                            cairo_lang_starknet_classes::casm_contract_class::CasmContractClass::from_contract_class(
-                                sierra_cc.clone(),
-                                false,
-                                usize::MAX,
-                            )
-                            .unwrap();
-                        let casm_cc: GenericCasmContractClass = casm_cc.into();
-
-                        log::debug!("class_hash (from RPC): {}", class_hash);
-
-                        // TODO: it seems that this ends up computing the wrong class hash...
-                        write_class_facts(
-                            sierra_cc,
-                            casm_cc.clone(),
-                            &mut initial_state.ffc.clone_with_different_hash::<PoseidonHash>(),
-                        )
-                        .await?;
-
-                        compiled_classes.insert(class_hash, casm_cc);
-                    }
-                    starknet::core::types::ContractClass::Legacy(_compressed_logacy_contract_class) => {
-                        panic!("legacy class (TODO)");
-                    }
-                }
-            } else {
-                log::warn!("No class available for contract {}", contract_address);
+            let generic_cc = generic_sierra_cc.compile()?;
+
+            log::debug!("============================");
+            log::debug!("class_hash (from RPC): {:x?}", class_hash);
+            log::debug!("class_hash (computed): {:x?}", generic_sierra_cc.class_hash()?);
+
+            // TODO: it seems that this ends up computing the wrong class hash...
+            let (contract_class_hash, compiled_contract_hash) = write_class_facts(
+                generic_sierra_cc,
+                generic_cc.clone(),
+                &mut initial_state.ffc.clone_with_different_hash::<PoseidonHash>(),
+            )
+            .await?;
+
+            class_hash_to_compiled_class_hash.insert(class_hash, compiled_contract_hash.into());
+            compiled_classes.insert(compiled_contract_hash.into(), generic_cc.clone());
+
+            log::debug!(
+                "Computed class_hash => compiled_class_hash: {:x?} => {:x?}",
+                contract_class_hash,
+                compiled_contract_hash
+            );
+
+            // let sierra_cc_hash = generic_sierra_cc.class_hash()?;
+            let sierra_cc_hash = generic_cc.class_hash()?;
+            log::debug!("sierra_cc_hash: {:x?}", sierra_cc_hash);
+
+            let class_hash_as_snos_hash: starknet_os_types::hash::Hash = class_hash.into();
+            if contract_class_hash != class_hash_as_snos_hash {
+                log::warn!("FFC computed different hash ({:x?}) than expected ({:?})", contract_class_hash, class_hash);
```

Comment:
What this means?

Reply:
I would remove it. Or make it panic.

Reply:
Removed this and other verbose class hash logging in https://github.com/Moonsong-Labs/snos/pull/19/commits/f032b2fc3de9b28cf6c0b690ea813e69b16d223c